### PR TITLE
fix: fixes DI scope disposing only when message processing ends

### LINKED ISSUE #225: BatchConsume is broken in version >= 2.1.4
There is a problem with `BatchConsume` after adding middleware instance lifetimes.
Middleware after `BatchConsumeMiddleware` is instantiated with disposed scope when postponed `next` is dispatched.

**Steps to reproduce:**
1. Run `KafkaFlow.Sample.BatchConsume` sample project
2. Send message and wait 10 seconds

**Exception message:**

> System.ObjectDisposedException: 'Cannot access a disposed object. Object name: 'IServiceProvider'.'

### PR DESCRIPTION
**Pull request by filipeesch**
# Description

fixes DI scope disposing only when message processing ends

Fixes #225 

## How Has This Been Tested?

Manually tested. Integration and Unit tests were executed

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [ ] I have added tests to cover my changes
-   [ ] I have made corresponding changes to the documentation

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: src/KafkaFlow.UnitTests/BatchConsume/BatchConsumeMiddlewareTests.cs [hunk 1]
```diff
@@ -2,94 +2,139 @@ namespace KafkaFlow.UnitTests.BatchConsume
 {
```
Comment (dougolima) [inline src/KafkaFlow.UnitTests/BatchConsume/BatchConsumeMiddlewareTests.cs:2]: What do you think about we create integration tests with batch produce/consume?

Comment (filipeesch) [reply src/KafkaFlow.UnitTests/BatchConsume/BatchConsumeMiddlewareTests.cs:2]: Yes we must create

File: src/KafkaFlow.BatchConsume/BatchConsumeMiddleware.cs [hunk 2]
```diff
@@ -1,39 +1,121 @@
 ﻿namespace KafkaFlow.BatchConsume
```
[the remark this answers is not in the document]
Comment (filipeesch) [reply src/KafkaFlow.BatchConsume/BatchConsumeMiddleware.cs:1]: renamed to BatchOperations

File: src/KafkaFlow.BatchConsume/BatchConsumeMiddleware.cs [hunk 3]
```diff
@@ -1,39 +1,121 @@
...
+            }
+
+            // stores the Task in a local variable to avoid NullReferenceException in case the schedule Task ends before the await below
+            var localTask = this.dispatchTask;
+
+            if (localTask != null)
+            {
+                await localTask.Unwrap().ConfigureAwait(false);
+            }
+        }
```
[the remark this answers is not in the document]
Comment (filipeesch) [reply src/KafkaFlow.BatchConsume/BatchConsumeMiddleware.cs:48]: Great suggestion. Changed!

**Review by dougolima — APPROVED** [review]

**Pull request merged**